{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "afd3dad5",
   "metadata": {},
   "outputs": [],
   "source": [
    "from delfos.market.session import Session\n",
    "from delfos.broker.portfolio import Portfolio\n",
    "from datetime import datetime\n",
    "from matplotlib import pyplot as plt\n",
    "import datetime as dt\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6beeaebf",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n"
     ]
    }
   ],
   "source": [
    "dt = datetime(2021, 7, 21)\n",
    "\n",
    "session = Session(period=6)\n",
    "#session = Session(date=dt, period=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "31e78129",
   "metadata": {},
   "outputs": [],
   "source": [
    "carteira = Portfolio(session)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "58bbd8d1",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PETR4: qtd = 91, preco = 16.80956043956044, valor inicial = 1529.6699999999998, valor atual = 2769.130027770996\n",
      "Rentabilidade: 225.99%\n",
      "Valor ganho: 2217.4599999999996\n",
      "\n",
      "BPAC11: qtd = 556, preco = 10.570179856115109, valor inicial = 5877.02, valor atual = 18398.040084838867\n",
      "Rentabilidade: 223.67%\n",
      "Valor ganho: 624.3299999999999\n",
      "\n",
      "AZUL4: qtd = 86, preco = 21.94046511627907, valor inicial = 1886.88, valor atual = 1497.2599868774414\n",
      "Rentabilidade: -20.65%\n",
      "Valor ganho: 0\n",
      "\n",
      "CVCB3: qtd = 66, preco = 15.2, valor inicial = 1003.1999999999999, valor atual = 196.6800012588501\n",
      "Rentabilidade: -80.39%\n",
      "Valor ganho: 0\n",
      "\n",
      "FLRY3: qtd = 57, preco = 22.75989974937343, valor inicial = 1297.3142857142857, valor atual = 900.6000108718872\n",
      "Rentabilidade: -22.75%\n",
      "Valor ganho: 101.64\n",
      "\n",
      "MYPK3: qtd = 66, preco = 15.3, valor inicial = 1009.8000000000001, valor atual = 934.5599899291992\n",
      "Rentabilidade: 5.12%\n",
      "Valor ganho: 126.94999999999999\n",
      "\n",
      "ARZZ3: qtd = 19, preco = 56.365789473684224, valor inicial = 1070.9500000000003, valor atual = 1528.3600463867188\n",
      "Rentabilidade: 47.07%\n",
      "Valor ganho: 46.730000000000004\n",
      "\n",
      "MGLU3: qtd = 178, preco = 23.01174157303371, valor inicial = 4096.09, valor atual = 594.5199847221375\n",
      "Rentabilidade: -85.33%\n",
      "Valor ganho: 6.32\n",
      "\n",
      "TSLA34: qtd = 66, preco = 25.504545454545458, valor inicial = 1683.3000000000002, valor atual = 2515.9199295043945\n",
      "Rentabilidade: 49.46%\n",
      "Valor ganho: 0\n",
      "\n",
      "AMZO34: qtd = 12, preco = 43.567499999999995, valor inicial = 522.81, valor atual = 369.59999084472656\n",
      "Rentabilidade: -29.31%\n",
      "Valor ganho: 0\n",
      "\n",
      "RDOR3: qtd = 59, preco = 36.0, valor inicial = 2124.0, valor atual = 2067.9499549865723\n",
      "Rentabilidade: 4.22%\n",
      "Valor ganho: 145.75\n",
      "\n",
      "ITSA4: qtd = 162, preco = 9.067607289829512, valor inicial = 1468.952380952381, valor atual = 1600.5600185394287\n",
      "Rentabilidade: 16.69%\n",
      "Valor ganho: 113.53\n",
      "\n",
      "BBAS3: qtd = 20, preco = 29.1, valor inicial = 582.0, valor atual = 950.9999847412109\n",
      "Rentabilidade: 89.57%\n",
      "Valor ganho: 152.27\n",
      "\n",
      "VALE3: qtd = 32, preco = 80.3465625, valor inicial = 2571.09, valor atual = 2143.679931640625\n",
      "Rentabilidade: -8.15%\n",
      "Valor ganho: 217.97\n",
      "\n",
      "BCFF11: qtd = 15, preco = 72.34, valor inicial = 1085.1000000000001, valor atual = 1074.3000411987305\n",
      "Rentabilidade: 12.08%\n",
      "Valor ganho: 141.90000000000003\n",
      "\n",
      "CASH3: qtd = 40, preco = 28.5, valor inicial = 1140.0, valor atual = 399.1999816894531\n",
      "Rentabilidade: -64.98%\n",
      "Valor ganho: 0\n",
      "\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "unsupported operand type(s) for *: 'NoneType' and 'int'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[1;32mIn [4]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m position \u001b[38;5;129;01min\u001b[39;00m carteira\u001b[38;5;241m.\u001b[39mpositions:\n\u001b[1;32m----> 2\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m: qtd = \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m, preco = \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m, valor inicial = \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m, valor atual = \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m.\u001b[39mformat(position, carteira\u001b[38;5;241m.\u001b[39mpositions[position]\u001b[38;5;241m.\u001b[39mquantity, carteira\u001b[38;5;241m.\u001b[39mpositions[position]\u001b[38;5;241m.\u001b[39mprice_per_share, carteira\u001b[38;5;241m.\u001b[39mpositions[position]\u001b[38;5;241m.\u001b[39mheld_shares_cost, \u001b[43mcarteira\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpositions\u001b[49m\u001b[43m[\u001b[49m\u001b[43mposition\u001b[49m\u001b[43m]\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheld_shares_value\u001b[49m))\n\u001b[0;32m      3\u001b[0m     \u001b[38;5;28mprint\u001b[39m( \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRentabilidade:\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{:,.2f}\u001b[39;00m\u001b[38;5;124m%\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mformat(carteira\u001b[38;5;241m.\u001b[39mpositions[position]\u001b[38;5;241m.\u001b[39mtotal_return \u001b[38;5;241m*\u001b[39m \u001b[38;5;241m100\u001b[39m))\n\u001b[0;32m      4\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mValor ganho:\u001b[39m\u001b[38;5;124m\"\u001b[39m, carteira\u001b[38;5;241m.\u001b[39mpositions[position]\u001b[38;5;241m.\u001b[39mtotal_income)\n",
      "File \u001b[1;32m~\\OneDrive\\Projetos\\Meus\\Stocks-Advisor\\delfos\\broker\\position.py:132\u001b[0m, in \u001b[0;36mPosition.held_shares_value\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    130\u001b[0m \u001b[38;5;129m@property\u001b[39m\n\u001b[0;32m    131\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mheld_shares_value\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[1;32m--> 132\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_held_shares_value\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\OneDrive\\Projetos\\Meus\\Stocks-Advisor\\delfos\\broker\\position.py:200\u001b[0m, in \u001b[0;36mPosition.get_held_shares_value\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    191\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget_held_shares_value\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[0;32m    192\u001b[0m     \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m    193\u001b[0m \u001b[38;5;124;03m    Cálcula o valor das ações atuais (não conta as vendidas/realizadas)\u001b[39;00m\n\u001b[0;32m    194\u001b[0m \n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    198\u001b[0m \u001b[38;5;124;03m        Valor das ações mantidas\u001b[39;00m\n\u001b[0;32m    199\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m--> 200\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m__stock\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcurrent_price\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m__quantity\u001b[49m\n",
      "\u001b[1;31mTypeError\u001b[0m: unsupported operand type(s) for *: 'NoneType' and 'int'"
     ]
    }
   ],
   "source": [
    "for position in carteira.positions:\n",
    "    print(\"{}: qtd = {}, preco = {}, valor inicial = {}, valor atual = {}\".format(position, carteira.positions[position].quantity, carteira.positions[position].price_per_share, carteira.positions[position].held_shares_cost, carteira.positions[position].held_shares_value))\n",
    "    print( \"Rentabilidade:\", '{:,.2f}%'.format(carteira.positions[position].total_return * 100))\n",
    "    print(\"Valor ganho:\", carteira.positions[position].total_income)\n",
    "    print()\n",
    "print()\n",
    "print(\"Valor gasto:\", '{:,.2f}'.format(carteira.total_cost))\n",
    "print()\n",
    "print(\"Valor ganho:\", '{:,.2f}'.format(carteira.total_income))\n",
    "print()\n",
    "print(\"Valor atual:\", '{:,.2f}'.format(carteira.held_shares_value))\n",
    "print()\n",
    "print(\"Diferença:\", '{:,.2f}'.format(carteira.total_change))\n",
    "print()\n",
    "print(\"Rentabilidade:\", '{:,.2f}%'.format(carteira.total_return*100))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e41c38a1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "carteira.positions\n",
    "\n",
    "rets_dict = {k: v for k, v in zip(carteira.positions.keys(), [v.total_return*100 for v in carteira.positions.values()])}\n",
    "{k: v for k, v in sorted(rets_dict.items(), key=lambda item: item[1], reverse=True)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "390a0e19",
   "metadata": {},
   "outputs": [],
   "source": [
    "carteira.failed_positions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "230bbff1",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "port_hist = carteira.portfolio_history\n",
    "\n",
    "#CONSERTOS DE DATAS AUSENTES NO HISTORICO BAIXADO DO YFINANCE\n",
    "port_hist = port_hist.drop(index=(datetime(2023, 3, 10))) #yfinance não manda o historico de BPAC11 no dia 10/03/2023\n",
    "#port_hist = port_hist.drop(index=(datetime(2023, 7, 27)))\n",
    "\n",
    "#port_hist[(port_hist[\"Cash Flow\"]> 0) | (port_hist[\"Cash Flow\"]< 0)]\n",
    "#port_hist[(port_hist.index.year == 2023) & (port_hist.index.month == 3)]\n",
    "port_hist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9658e972",
   "metadata": {},
   "outputs": [],
   "source": [
    "#ANÁLISE DO PORTFOLIO CONTRA O MERCADO E A RENDA FIXA\n",
    "\n",
    "\n",
    "market_hist = session.market_index.history[\"Close\"][session.market_index.history.index.date >= carteira.first_date]\n",
    "market_hist = market_hist.drop(index=(datetime(2023, 3, 10))) #yfinance não manda o historico de BPAC11 no dia 10/03/2023\n",
    "\n",
    "\n",
    "free_risk_hist = session.risk_free_rate_series\n",
    "free_risk_hist = free_risk_hist[free_risk_hist.index.date >= carteira.first_date]\n",
    "free_risk_hist = free_risk_hist \n",
    "\n",
    "\n",
    "plt.plot(port_hist[\"Total Ret\"]*100, label=\"Portfolio\")\n",
    "plt.plot((market_hist.pct_change().add(1).cumprod().sub(1) * 100), label=\"Ibovespa\")\n",
    "plt.plot((free_risk_hist.cumsum()* 100), label=\"Selic\")\n",
    "\n",
    "\n",
    "#add axes labels and a title\n",
    "plt.legend()\n",
    "\n",
    "plt.ylabel('Rentabilidade (%)', fontsize=14)\n",
    "plt.title('Performance do Portfolio', fontsize=16)\n",
    "\n",
    "plt.xticks(rotation=20)\n",
    "\n",
    "#plt.margins(0.2)\n",
    "  \n",
    "# Tweak spacing to prevent clipping of tick-labels\n",
    "plt.subplots_adjust(bottom = 0.1)\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e5ce92da",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-484:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-152:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Exception in thread Thread-207:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-319:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-198:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-422:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Exception in thread Thread-119:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Exception in thread Thread-348:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-28:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-156:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-369:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "        self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "        raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "        self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-345:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-479:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-465:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-443:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-391:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-232:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-145:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Exception in thread Thread-287:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-296:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-105:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-237:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "Exception in thread Thread-193:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-217:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-183:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-338:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-191:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "        self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-173:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Exception in thread Thread-327:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-138:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-294:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-297:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Connection to remote host was lost.\n",
      "Exception in thread Thread-58:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Connection to remote host was lost.\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-276:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-113:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-83:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "Exception in thread Thread-259:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-241:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-388:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-212:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-500:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-501:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-245:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-221:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-337:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-365:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-379:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-272:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-283:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-40:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-485:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Connection to remote host was lost.\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-432:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-394:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-448:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-424:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Exception in thread Thread-393:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Thread-433:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-444:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-543:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-291:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-53:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-127:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-121:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-73:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-184:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-108:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Exception in thread Thread-209:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-84:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-312:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-279:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-78:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-34:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5C205E0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB5B8FDC0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "error from callback <function Stock.__set_price_streamer.<locals>.<lambda> at 0x0000021BB4F504C0>: '>' not supported between instances of 'NoneType' and 'float'\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-392:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-517:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-423:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-368:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "Exception in thread Thread-473:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Exception in thread Thread-399:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "        self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Connection to remote host was lost.\n",
      "Exception in thread Thread-492:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-418:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-435:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-525:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-532:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-390:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-376:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-247:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-286:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-243:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-210:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "Exception in thread     self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "Thread-228:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-202:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "Exception in thread Thread-144:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-110:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-204:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-290:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "Exception in thread Thread-134:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-130:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n",
      "Connection to remote host was lost.\n",
      "'SSLDispatcher' object has no attribute 'timeout'\n",
      "Exception in thread Thread-238:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 910, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 125, in _run\n",
      "    self.stop()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\yflive\\streamer.py\", line 113, in stop\n",
      "    self._ws_thread.join()\n",
      "  File \"C:\\Users\\bgalv\\AppData\\Local\\Programs\\Python\\Python39\\lib\\threading.py\", line 1050, in join\n",
      "    raise RuntimeError(\"cannot join current thread\")\n",
      "RuntimeError: cannot join current thread\n"
     ]
    }
   ],
   "source": [
    "print(\"Rentabilidades (TOTAL):\\n\")\n",
    "\n",
    "print(\"Ibov: \", round(float( (market_hist.pct_change().add(1).cumprod().sub(1)).iloc[-1]),4)*100, \"%\")\n",
    "print(\"Portfolio: \", round(carteira.total_return, 4)*100, \"%\")\n",
    "print(\"Selic: \", round(float(free_risk_hist.cumsum().iloc[-1]), 4)*100, \"%\")\n",
    "\n",
    "port_cdi = round(float((carteira.total_return * 100) / free_risk_hist.cumsum().iloc[-1]), 2)\n",
    "\n",
    "print(\"Portfolio (CDI): \", port_cdi, \"%\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d22f2a37",
   "metadata": {},
   "outputs": [],
   "source": [
    "pos = carteira.positions[\"BPAC11\"]\n",
    "#pos.history[(pos.history.index.year == 2023) & (pos.history.index.month == 3)]\n",
    "pos.history\n",
    "#pos.history[pos.history[\"Cash Flow\"]> 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30287045",
   "metadata": {},
   "outputs": [],
   "source": [
    "stock = session.select_stocks(\"BPAC11\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b765e5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "stock.history"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed5e9392",
   "metadata": {},
   "outputs": [],
   "source": [
    "stock.current_price"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46ee0acf",
   "metadata": {},
   "source": [
    "### Testes com o Portfolio Total (cash.json)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f4d88fe",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#--------------------------------- TESTES COM PORTFOLIO TOTAL (CASH_FILE) ----------------------------------------------\n",
    "\n",
    "\n",
    "import pandas as pd\n",
    "import json\n",
    "\n",
    "with open(\"./delfos/data/broker/cash.json\", \"r\") as cash_file:\n",
    "    cash_data = json.load(cash_file)\n",
    "    \n",
    "net_data_hist = []\n",
    "for data in cash_data:\n",
    "    date = data[\"date\"]\n",
    "    available_cash = data[\"available_cash\"]\n",
    "    net_data = data[\"net_worth\"]\n",
    "    \n",
    "    if net_data[\"broker\"] < 0:\n",
    "        net_data[\"broker\"] = 0\n",
    "        \n",
    "    net_data.update({\"date\": date})\n",
    "    net_data_hist.append(net_data)\n",
    "\n",
    "df = pd.DataFrame(net_data_hist).set_index(\"date\")\n",
    "df.index = pd.to_datetime(df.index, format=\"%d-%m-%Y\")\n",
    "\n",
    "df = df.merge(carteira.portfolio_history[\"Close (no income)\"], left_index=True, right_index=True, how=\"outer\")\n",
    "\n",
    "df = df.rename(columns={\"Close (no income)\": \"stocks\"})\n",
    "\n",
    "df = df.fillna(method='bfill')\n",
    "df = df.fillna(method='ffill')\n",
    "#df = df.dropna()\n",
    "\n",
    "df[\"total\"] = df.sum(axis=1)\n",
    "df[\"total_stocks\"] = df[\"stocks\"] + df[\"stocks_funds\"]\n",
    "df[\"total_fixed\"] = df[\"bank\"] + df[\"savings\"] + df[\"public_bonds\"]\n",
    "\n",
    "df = df.drop(index=(datetime(2023, 3, 10))) #yfinance não manda o historico de BPAC11 no dia 10/03/2023"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5106b0e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04afe812",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df[\"total\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d68db54c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"total\"].plot()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "397dc533",
   "metadata": {},
   "outputs": [],
   "source": [
    "for col in df.columns[:-3]:\n",
    "    pct = df.iloc[-1][col] / df.iloc[-1][\"total\"]\n",
    "    print(col + \": \" + str(round(pct*100, 2)) + \"%\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98e3c72b",
   "metadata": {},
   "outputs": [],
   "source": [
    "stocks_pct = df.iloc[-1][\"total_stocks\"] / df.iloc[-1][\"total\"]\n",
    "fixed_pct = df.iloc[-1][\"total_fixed\"] / df.iloc[-1][\"total\"]\n",
    "cash_pct = df.iloc[-1][\"broker\"] / df.iloc[-1][\"total\"]\n",
    "\n",
    "print(\"Stocks: \", round(stocks_pct * 100, 2), \"%\")\n",
    "print(\"Fixed Income: \", round(fixed_pct * 100, 2), \"%\")\n",
    "print(\"Cash: \", round( cash_pct * 100, 2), \"%\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
